# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/IMEData.h
# Commit: 88cd68e70502
# Full Hash: 88cd68e7050272bf76268c904feef7348d257b56
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-02-08 17:32:27
# Regressor Bug: 1746104
# File Overlap Count: 1
# Description:
#   Bug 1746104 - part 6-2: Make constructors of `ContentCache::Selection` take `IMENotification::SelectionChangeData` or `WidgetQueryContentEvent` r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137428
# ==============================================================================

diff -r 2529c39ed6bf -r 88cd68e70502 widget/IMEData.h
--- a/widget/IMEData.h	Mon Feb 07 22:33:39 2022 +0000
+++ b/widget/IMEData.h	Mon Feb 07 22:33:40 2022 +0000
@@ -767,6 +767,14 @@
     void SetWritingMode(const WritingMode& aWritingMode);
     WritingMode GetWritingMode() const;
 
+    uint32_t StartOffset() const {
+      MOZ_ASSERT(mHasRange);
+      return mOffset;
+    }
+    uint32_t EndOffset() const {
+      MOZ_ASSERT(mHasRange);
+      return mOffset + Length();
+    }
     uint32_t AnchorOffset() const {
       MOZ_ASSERT(mHasRange);
       return mOffset + (mReversed ? Length() : 0);